adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/FeatureRDD.scala

```diff
@@ -121,11 +113,13 @@ object FeatureRDD {
     // cache the rdd, since we're making multiple passes
     rdd.cache()
 
-    // aggregate to create the sequence dictionary
-    val sd = new SequenceDictionary(rdd.map(getSequenceRecord)
-      .distinct
-      .collect
-      .toVector)
+    // create sequence records based on largest end coordinate
+    val featuresByContigName = rdd.keyBy(_.getContigName)
```

Comment:
Not sure this is a good idea, but it (partially) solves the trying to partition a sequence of length 1L problem in `GenomicRDD.pipe`